### PR TITLE
reimplement random ChangeId generator without using UUID

### PR DESCRIPTION
**Pull request by yuja**
`create_no_gc_ref()` is the last remaining dependency on UUID.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: lib/src/settings.rs [hunk 1]
```diff
@@ -155,13 +155,11 @@ impl UserSettings {
...
-        ChangeId::from_bytes(
-            uuid::Builder::from_random_bytes(random_bytes)
-                .into_uuid()
-                .as_bytes(),
-        )
+        let mut random_bytes: [u8; 16] = self.gen();
+        // TODO: make it fully random
+        random_bytes[6] = (random_bytes[6] & 0x0f) | ((4 as u8) << 4); // Version: 4
+        random_bytes[8] = (random_bytes[8] & 0x3f) | 0x80; // Variant::RFC4122
+        ChangeId::new(random_bytes.into())
```
Comment (martinvonz) [inline lib/src/settings.rs:159]: Can we remove the dependency on the `uuid` crate now then?

**Review by martinvonz — APPROVED** [review]

File: lib/src/settings.rs (hunk 1 above)
Comment (yuja) [reply lib/src/settings.rs:159]: Sadly there's one more dep: `create_no_gc_ref()`. Does hyphenated syntax matter there?

Comment (martinvonz) [reply lib/src/settings.rs:159]: It doesn't matter for any technical reason and I don't think users care either. Feel free to change it as far as I'm concerned. But also feel free to merge this as is.

Comment (yuja) [reply lib/src/settings.rs:159]: I'll send another PR to drop uuid dependency.

**Pull request merged**